src/coreclr/jit/importercalls.cpp

```diff
@@ -2815,6 +2815,10 @@ GenTree* Compiler::impCreateSpanIntrinsic(CORINFO_SIG_INFO* sig)
 
     CORINFO_CLASS_HANDLE fieldOwnerHnd = info.compCompHnd->getFieldClass(fieldToken);
 
+    // We're expanding RuntimeHelpers.CreateSpan<int>((RuntimeFieldHandle)0x...), thus, we need to
+    // statically initialize the class containing that RuntimeFieldHandle
```

Comment:
I am not following why we need to do this. The comment should explain why we need to do this.

This comment suggests that the non-expanded version of RuntimeHelpers.CreateSpan performs equivalent operation, but I do not see an equivalent operation in non-expanded version of RuntimeHelpers.CreateSpan. I am not sure whether this is the right fix.

Reply:
Case 1: `ReadOnlySpan<byte>` RVA: [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGAzAATakDChA3oaQ+TkqQEowCGAJgPIB2AGwCeAZQAOHPgB5gQgC4wAfKQAiHOR1IBeZQG0cAGiwHiBlAF0A3PUYZmpWQtIAVGAGc5ACgCU25Wo19K0IAXyA===)

here JIT triggers class initialization for `<PrivateImplementationDetails>` when it imports `get_Data()` method and sees `ldsflda` for which it calls `getFieldInfo` and triggers class init in jit time [here](https://github.com/dotnet/runtime/blob/f8bcb89796d3c83264fffd913e5196d29d8d730e/src/coreclr/jit/importer.cpp#L9143)

Nothing like that happens for `CreateSpan` case - there is no ld[s]fld opcode anywhere, only `ldtoken field`

> This comment suggests that the non-expanded version of RuntimeHelpers.CreateSpan performs equivalent operation,

Then I guess it should do the same somehow?

Reply:
Maybe David's PR indeed broke this behavior, but it seems that if we put some real cctor to `<PrivateImplementationDetails>` - nobody will call it

Reply:
> Then I guess it should do the same somehow?

`RuntimeHelpers.CreateSpan` and `RuntimeHelpers.InitializeArray` APIs never triggered static constructors. They are special APIs. Nothing says that they have to trigger static constructor. I do not see a problem with it.